meson.build

```diff
@@ -304,6 +304,24 @@ dir_data_extension = dir_data / 'extension'
 
 
 
+###############################################################
+# Library: bonjour
+###############################################################
+
+bonjouropt = get_option('bonjour')
+if not bonjouropt.disabled()
+  bonjour = dependency('', required: bonjouropt)
```

Comment:
Not sure what the 'required: bonjouropt' is supposed to do for the '' dependency?

Reply:
If `bonjour=enabled` this will error out with `ERROR: Dependency is required but has no candidates.`

Moreover I don't understand the code at all? The entire support code for bonjour amounts to:

- check if `dns_sd.h` exists, if so, set `USE_BONJOUR` in the configuration data
- declare a bonjour dependency() object twice, under the following conditions:
  - if bonjour is disabled, do not create the variable at all; attempting to add it to a target *may* be an undefined variable error
  - the first time, as a not-found-dependency (only if bonjour is enabled or auto)
  - the second time, overwrite it as an internal declare_dependency() containing no compile/link instructions (only if the header is found)
- do not use it other than to print a summary; it may be an undefined variable error, otherwise it will be a very fancy boolean

Advice: just replace this all with:

```meson
bonjouropt = get_option('bonjour')
bonjour_found = cc.check_header('dns_sd.h', args: g_c_args, required: bonjouropt)
if bonjour_found
    cdata.set('USE_BONJOUR', 1)
endif

...
...

  summary(
    {
      'bonjour': bonjour_found, # bool
...
```

Reply:
Thanks for the review! I guess it's nice to use declare_dependency() for bonjour, for symmetry reasons (and for when we add fallback to avahi etc). The rest of what you said obviously still applies...

Reply:
Thanks for the reviews! I updated the code.